### DISCUSSION TITLE
Allow user to set a certain commit body when enabling auto-merge on a PR

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#24907] [Suggest an Idea]
### What would you like Renovate to be able to do?

**Summary:** a way to tell Renovate to set a certain commit body when enabling auto-merge on a PR.

Relevant quote from @AlexanderWirthSAP:

> So I don't want to remove information from the prBody, I am looking for a way to tell Renovate to set a certain commit body when enabling auto-merge on a PR. Based on your replies, it seems that this is currently not possible. I think it's a reasonable feature request though, given my example of the current behavior above. All I would need is some new setting like for example "automergeCommitBody" that lets me control the commit body that Renovate should use when enabling auto-merge.

### Current behavior

> ```
> commit 8524bffde3aeb3144d720d950f3bc1317e59ea41 (HEAD -> master, origin/master)
> Author: ospo-renovate[bot] <84706+ospo-renovate[bot]@users.noreply.github.wdf.sap.corp>
> Date:   Thu Apr 6 14:16:21 2023 +0200
> 
>     Update dependency com.sap.sgs:security-patched-spring-boot-starter to v3.0.5-sap-01 (#38)
> 
>     This PR contains the following updates:
> 
>     | Package | Type | Update | Change |
>     |---|---|---|---|
>     | [com.sap.sgs:security-patched-spring-boot-starter](https://spring.io/projects/spring-boot) ([source](https://github.wdf.sap.corp/Phosphor/security-patched-spring-boot-30x)) | parent | patch | `3.0.4-sap-01` -> `3.0.5-sap-01` |
> 
>     ---
> 
>     ### Release Notes
> 
>     <details>
>     <summary>Phosphor/security-patched-spring-boot-30x</summary>
> 
>     ### [`v3.0.5-sap-01`](https://github.wdf.sap.corp/Phosphor/security-patched-spring-boot-30x/blob/HEAD/CHANGELOG.md#&#8203;305-sap-01---2023-03-23)
> 
>     ##### Update
> 
>     -   Updated to `org.springframework.boot:spring-boot-starter-parent:3.0.5`
>         and `org.springframework.boot:spring-boot-dependencies:3.0.5`.
> 
>     ##### Security
> 
>     -   Updated `org.codehaus.jettison:jettison` from `1.5.2` to `1.5.4`
>         to fix [CVE-2023-1436](https://nvd.nist.gov/vuln/detail/CVE-2023-1436).
> 
>     ### [`v3.0.4-sap-02`](https://github.wdf.sap.corp/Phosphor/security-patched-spring-boot-30x/blob/HEAD/CHANGELOG.md#&#8203;304-sap-02---2023-03-10)
> 
>     ##### Security
> 
>     -   Updated `org.wildfly.security:wildfly-elytron` from `1.20.1.Final` to `1.20.3.Final` which inturn updates `org.apache.sshd:sshd-common` from `2.7.0` to `2.9.2`
>         to fix [CVE-2022-45047](https://nvd.nist.gov/vuln/detail/CVE-2022-45047).
> 
>     </details>
> 
>     ---
> 
>     ### Configuration
> 
>     📅 **Schedule**: Branch creation - At any time (no schedule defined), Automerge - At any time (no schedule defined).
> 
>     🚦 **Automerge**: Enabled.
> 
>     ♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the rebase/retry checkbox.
> 
>     🔕 **Ignore**: Close this PR and you won't be reminded about this update again.
> 
>     ---
> 
>      - [ ] If you want to rebase/retry this PR, check this box
> 
>     ---
> 
>     [Trigger Renovate Run](https://renovate-wdf.internal.opensource.tools.sap/run/LIS/renovate-test) | [View Renovate Logs](https://renovate-wdf.internal.opensource.tools.sap/logs/LIS/renovate-test)
>     Generated by [Renovate Bot](https://github.com/renovatebot/renovate) and managed by the [TGIF-project renovate-controller](https://github.tools.sap/OSPO/renovate-controller).
>     For questions/issues please raise an issue in the renovate-controller repository.
>     
> 
>     Co-authored-by: OSPO Renovate Bot <84706+ospo-renovate[bot]@users.noreply.github.wdf.sap.corp>
> ```

### Desired behavior

> ```
> commit 8524bffde3aeb3144d720d950f3bc1317e59ea41 (HEAD -> master, origin/master)
> Author: ospo-renovate[bot] <84706+ospo-renovate[bot]@users.noreply.github.wdf.sap.corp>
> Date:   Thu Apr 6 14:16:21 2023 +0200
> 
>     Update dependency com.sap.sgs:security-patched-spring-boot-starter to v3.0.5-sap-01 (#38)
> ```

### If you have any ideas on how this should be implemented, please tell us here.

@viceice said: [^quote]

> that sounds good to me. it should be `undefined` by default to not break existing users.

[^quote]: https://github.com/renovatebot/renovate/discussions/21369#discussioncomment-5555238

### Is this a feature you are interested in implementing yourself?

No

**Comment by viceice** [comment]
i think this is GitHub specific

**Reply by philippe-granet** [reply]
We have the same problem on Gitlab, but I dont find any Gitlab API to update merge_commit_message at MR creation or update, i can only update it on Gitlab UI
